sys/net/gnrc/application_layer/uhcpc/gnrc_uhcpc.c

```diff
@@ -54,6 +57,32 @@ static void set_interface_roles(void)
 
 static ipv6_addr_t _prefix;
 
+#ifdef MODULE_GNRC_SIXLOWPAN_CTX
+#define SIXLO_CTX_LTIME (60)
+
+static void _update_prefix(const ipv6_addr_t *prefix, uint8_t prefix_len)
+{
+    gnrc_sixlowpan_ctx_t *ctx;
+    uint8_t cid = 0;
+
+    /* find first free context ID */
+    while (((ctx = gnrc_sixlowpan_ctx_lookup_id(cid)) != NULL) &&
+           ((ctx->prefix_len != prefix_len) ||
+            (ipv6_addr_match_prefix(&ctx->prefix, prefix) < prefix_len))) {
+        cid++;
+    }
```

Comment:
I strongly suspect this breaks if I have a old prefix at `CID 0` and my new prefix is at `CID 1`. When the old prefix times out, the new prefix will get a new allocation at `CID 0` in addition to the one at `CID 1`

Reply:
While this is broken indeed, it has the nice benefit that CID 0 does not require an extra byte in the compression header :sweat_smile:

Reply:
I was able to trigger the following state when abusing #13471 a bit:
```Shellsession
> 6ctx
cid|prefix                                     |C|ltime
-----------------------------------------------------------
  0|                         2001:db8:b:c::/64 |1|   60min
  1|                         2001:db8:b:c::/64 |1|   59min
```

Reply:
(this needs to be fixed within `gnrc_sixlowpan_ctx` though)

Reply:
Shouldn't the code first iterate over the existing allocated CIDs to check if the received prefix is in use in one of the slots? And then find the first empty CID if the received prefix is not already present?

Reply:
This is not the same issue as described in #13471 unfortunately :)

Reply:
I just noticed: its not a bug in `gnrc_sixlowpan_ctx`, but a usage error here indeed. The context is always provided with the update call, so the `gnrc_sixlowpan_ctx` can't do anything but comply. Will fix.

Reply:
Fixed!

Reply:
Is indeed fixed